go/vt/vtgate/planbuilder/builder.go

```diff
@@ -105,17 +107,41 @@ func BuildFromStmt(query string, stmt sqlparser.Statement, vschema ContextVSchem
 	return plan, nil
 }
 
+func getConfiguredPlanner(vschema ContextVSchema) selectPlanner {
+	switch vschema.Planner() {
+	case V3:
+		return buildSelectPlan
+	case Gen4, Gen4Left2Right, Gen4GreedyOnly:
+		return gen4Planner
+	case Gen4WithFallback:
+		fp := &fallbackPlanner{
+			primary:  buildSelectPlan,
+			fallback: gen4Planner,
+		}
+		return fp.plan
+	default:
+		// fail - unknown planner
+		return func(query string) func(sqlparser.Statement, ContextVSchema) (engine.Primitive, error) {
+			return func(sqlparser.Statement, ContextVSchema) (engine.Primitive, error) {
+				return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "unknown planner selected %d", vschema.Planner())
+			}
+		}
+	}
+}
+
 func buildRoutePlan(stmt sqlparser.Statement, vschema ContextVSchema, f func(statement sqlparser.Statement, schema ContextVSchema) (engine.Primitive, error)) (engine.Primitive, error) {
 	if vschema.Destination() != nil {
 		return buildPlanForBypass(stmt, vschema)
 	}
 	return f(stmt, vschema)
 }
 
-func createInstructionFor(query string, stmt sqlparser.Statement, vschema ContextVSchema) (engine.Primitive, error) {
+type selectPlanner func(query string) func(sqlparser.Statement, ContextVSchema) (engine.Primitive, error)
+
+func createInstructionFor(query string, stmt sqlparser.Statement, vschema ContextVSchema, sel selectPlanner) (engine.Primitive, error) {
```

Comment:
I think we do not need to pass selectPlanner here and `getConfiguredPlanner(vschema)` can be used directly.